### ISSUE TITLE
Extending preact's component prototype breaks expected behavior

### ISSUE THREAD

**Issue by s3ththompson** [choojs/choop#15]
Extending preact's component prototype without setting the constructor prevents [`state`, `context`, and `props`](https://github.com/developit/preact/blob/master/src/component.js#L23-L39) from being available in the constructor.

Is there any reason not to just export `require('preact').Component` directly?

**Comment by jongacnik** [comment]
@s3ththompson not able to look into it this second, but this was done as a convenience thing initially so you could get choo emit directly w/in a component: https://github.com/choojs/choop/pull/10

**Comment by s3ththompson** [comment]
@jongacnik I'm not super familiar with this pattern... why can't you set `require('rooch/component').prototype.emit ` if `rooch/component` is just this?
```js
module.exports = require('preact').Component
```

**Comment by jongacnik** [comment]
@s3ththompson Sorry this has taken a backburner! I want to take a closer look at this thought since will likely be dropping Choop into a new build over here. 

I believe I did it this way so as not to alter the original Preact Component class. So sorta like, create our own special copy of it.

**Comment by jongacnik** [comment]
@s3ththompson You are right about this one. Need to think of an alt way to extend the class... I don't think directly extending the prototype is the right move? 

This is a good one to solve as it has ramifications for https://github.com/jongacnik/monoapp-react as well

**Comment by jongacnik** [comment]
Fixed https://github.com/choojs/choop/pull/17
